test/python/utils.py

```diff
@@ -20,11 +28,271 @@ def tarantool_version():
+    without_rocks = {x for x in files_list if not x.startswith('.rocks')}
+
+    file_tree = original_file_tree
+    if skip_tarantool_binaries or not tarantool_enterprise_is_used():
+        file_tree = {x for x in file_tree if x not in ['tarantool', 'tarantoolctl']}
+
+    assert file_tree == without_rocks
+    assert all(x in files_list for x in original_rocks_content)
+
+
+def assert_filemode(filepath, filemode):
+    filepath = os.path.join('/', filepath)
+
+    if filepath == os.path.join('/usr/share/tarantool/', project_name, 'VERSION'):
+        assert filemode & 0o777 == 0o644
+    elif filepath.startswith('/etc/systemd/system/'):
+        assert filemode & 0o777 == 0o644
+    elif filepath.startswith('/usr/lib/tmpfiles.d/'):
+        assert filemode & 0o777 == 0o644
+    elif filepath.startswith('/usr/share/tarantool/'):
+        # a+r for files, a+rx for directories
+        required_bits = 0o555 if os.path.isdir(filepath) else 0o444
+        assert filemode & required_bits == required_bits
+
+
+def assert_filemodes(basedir):
+    known_dirs = set([
+        'etc', 'etc/systemd', 'etc/systemd/system',
+        'usr', 'usr/share', 'usr/share/tarantool',
+        'usr/lib', 'usr/lib/tmpfiles.d'
+    ])
+    filenames = recursive_listdir(basedir) - known_dirs
+
+    for filename in filenames:
+        # we don't check fileowner here because it's set in postinst script
+
+        # check filemode
+        if filename.startswith(os.path.join('usr/share/tarantool/', project_name, '.rocks')):
+            continue
+
+        # get filestat
+        file_stat = os.stat(os.path.join(basedir, filename))
+        filemode = file_stat.st_mode
+        assert_filemode(filename, filemode)
+
+
+def validate_version_file(distribution_dir):
+    original_keys = [
+        'TARANTOOL',
+        project_name,
+        # default app dependencies
+        'luatest',
+        'cartridge',
+        # known cartridge dependencies
+        'membership',
+        'checks',
+        'vshard',
+        'http',
+        'frontend-core',
+    ]
+
+    if tarantool_enterprise_is_used():
+        original_keys.append('TARANTOOL_SDK')
+
+    version_filepath = os.path.join(distribution_dir, 'VERSION')
+    assert os.path.exists(version_filepath)
+
+    version_file_content = {}
+    with open(version_filepath, 'r') as version_file:
+        file_lines = version_file.read().strip().split('\n')
+        for l in file_lines:
+            m = re.match(r'^([^=]+)=([^=]+)$', l)
+            assert m is not None
+
+            key, version = m.groups()
+            version_file_content[key] = version
+
+    for key in original_keys:
+        assert key in version_file_content
+
+
+def assert_files_mode_and_owner_rpm(filename):
+    DIRNAMES_TAG = 1118
+    DIRINDEXES_TAG = 1116
+
+    expected_tags = [
+        'basenames', DIRNAMES_TAG, DIRINDEXES_TAG, 'filemodes',
+        'fileusername', 'filegroupname'
+    ]
+
+    rpm = rpmfile.open(filename)
```

Comment:
You don't close this file

You can use `with rpmfile.open()...`

Reply:
Yes, you right, thanks!